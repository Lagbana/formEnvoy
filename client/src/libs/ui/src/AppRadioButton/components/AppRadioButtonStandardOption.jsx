import * as React from 'react';
import { TouchableOpacity, StyleSheet, View, Text } from 'react-native';
import { Typography, Colors, Layouts } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppRadioButtonDirection, AppRadioButtonOptionStandardType } from '../AppRadioButton.interface';
import { AppAnimatedView } from '../../AppAnimatedView/AppAnimatedView';
import { AppIcon } from '../../AppIcon';

export interface AppRadioButtonStandardOptionProps<T> {
	option: AppRadioButtonOptionStandardType<T>;
	selectedValue?: T | null;
	direction: AppRadioButtonDirection;
	onSelect?: (value: T) => void;
	animation: {
		translateY: number;
		opacity: number;
	};
}

export const AppRadioButtonStandardOption: <T = any>(props: AppRadioButtonStandardOptionProps<T>) => React.ReactElement = (props) => {
	const { 
		direction,
		selectedValue,
		onSelect,
		option: { value, icon, image, label }, 
		animation: { translateY, opacity } 
	} = props;

	const styles = StyleSheet.create({
		box: {
      width: direction === 'row' ? '50%' : '100%',
      padding: 7,
		},
		image: {
      width: '100%',
      aspectRatio: 1,
      borderBottomRightRadius: 14,
      borderBottomLeftRadius: 14,
		},
		default: {
			...Layouts.shadowTile,
			backgroundColor: Colors.white,
			flex: 1,
		},
		wrapper: {
			flexDirection: 'row',
			alignItems: 'center',
			paddingVertical: 20,
			paddingLeft: direction === 'row' ? 10 : 20,
			paddingRight: direction === 'row' ? 10 : 50,
			flexShrink: 1,
			flex: 1,
		},
		selected: {
			backgroundColor: Colors.green,
		},
		label: {
			...Typography.labelDefault,
			color: Colors.purple,
			flexShrink: 1,
		},
		selectedLabel: {
			color: Colors.white,
		},
		icon: {
			color: Colors.white,
			marginLeft: direction === 'row' ? 10 : 20,
		},
	});

	const handleSelect = () => { if (onSelect) { onSelect(value); } }

	const isActive = selectedValue === value;
	const iconDisplay = (icon !== '' && icon !== undefined) ? <AppIcon
		name={icon || 'check'}
		size={22}
		style={[
			styles.icon,
			{ opacity: isActive ? 1 : 0 },
		]}
	/> : null;

	const imageDisplay = (image) ? <View style={styles.image}>{image}</View> : null;

	return (
		<AppAnimatedView
			style={
				[
					styles.box,
					/*
					{ 
						opacity,
						transform: [
							{
								translateY
							}
						]
					}
					*/
				]
			}>
				<TouchableOpacity
					style={[
						styles.default,
						isActive ? styles.selected : {},
					]}
					onPress={handleSelect}
					onLongPress={handleSelect}
				>
					<View style={styles.wrapper}>
						<Text
							style={[
								styles.label,
								isActive ? styles.selectedLabel : {}
							]}
						>
							{label}
						</Text>
						{iconDisplay}
					</View>
					{imageDisplay}
				</TouchableOpacity>
			</AppAnimatedView>
		);
			
}