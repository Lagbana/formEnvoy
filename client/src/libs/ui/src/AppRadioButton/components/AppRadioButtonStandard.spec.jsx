import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppRadioButtonStandardOption } from './AppRadioButtonStandardOption'

configure({ adapter: new Adapter() })

test('<AppRadioButtonStandardOption /> renders correctly', () => {
  const component = shallow(
    <AppRadioButtonStandardOption
      animation={{ opacity: 100, translateY: 0 }}
      option={{ value: 0, label: 'test label' }}
      direction={'row'}
    />
  )
  expect(toJson(component)).toMatchSnapshot()
})
