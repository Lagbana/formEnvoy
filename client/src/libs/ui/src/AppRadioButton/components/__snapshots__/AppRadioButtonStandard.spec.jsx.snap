// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRadioButtonStandardOption /> renders correctly 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "padding": 7,
        "width": "50%",
      },
    ]
  }
>
  <ForwardRef
    onLongPress={[Function]}
    onPress={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 14,
          "elevation": 20,
          "flex": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 15,
            "width": 0,
          },
          "shadowOpacity": 0.16,
          "shadowRadius": 30,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexShrink": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#4749A0",
              "flexShrink": 1,
              "fontFamily": "ObjektivMk1-Bold",
              "fontSize": 18,
              "lineHeight": 19.8,
              "paddingTop": 4.5,
            },
            Object {},
          ]
        }
      >
        test label
      </Text>
    </View>
  </ForwardRef>
</ForwardRef>
`;
